src/adcm_pytest_plugin/docker_utils.py

```diff
@@ -325,6 +329,33 @@ def __init__(self, container: Container, ip: str, port: str, container_config: O
     def stop(self):
         """Stops container"""
         self.container.stop()
+        with suppress(NotFound):
+            condition = "removed" if self.container.attrs["HostConfig"]["AutoRemove"] else "stopped"
+            self.container.wait(condition=condition, timeout=30)
+
+    @allure.step("Upgrade adcm to {target}")
+    def upgrade(self, target: Tuple[str, str]) -> None:
+        image, tag = target
+        assert self.container_config.volumes, "has no volume to move data"
+        volume_name = list(self.container_config.volumes.keys()).pop()
+        volume = self.container_config.volumes.get(volume_name)
+        with allure.step("Copy /adcm/data to folder attached by volume"):
+            self.container.exec_run(f"rsync -raAXH /adcm/data/ {volume['bind']}")
+        with allure.step("Update container config"):
+            self.container_config.image = image
+            self.container_config.tag = tag
+            volume.update({"bind": "/adcm/data"})
+            allure.attach(
+                json.dumps(asdict(self.container_config), indent=2),  # config object is not serializable
+                name="Container config for upgrade",
+                attachment_type=AttachmentType.JSON,
+            )
+        with allure.step("Start newer adcm"):
+            self.stop()
```

Comment:
Wrap this call into the separate step "Stop old container"

Reply:
Done.